test/libsolidity/SemanticTest.cpp

```diff
@@ -98,11 +98,18 @@ TestCase::TestResult SemanticTest::run(ostream& _stream, string const& _linePref
 			}
 			else
 			{
-				bytes output = callContractFunctionWithValueNoEncoding(
-					test.call().signature,
-					test.call().value,
-					test.call().arguments.rawBytes()
-				);
+				bytes output;
+				if (!test.call().hasCalldata)
```

Comment:
This flag as well - couldn't this just be ``test.call().signature.empty()`` instead?

Reply:
This flag is needed, since the signature is not empty. It will be `()` in case no name (identifier) was found and I think this is cleaner then just a string comparison. The function that parses the signature will throw if no name was found but parameters were given.

Reply:
I'd say string comparison is better than duplicating the information... but I'm fine with keeping the flag as well.

Reply:
Actually also: if you actually throw, if the signature is not found, then we need to explicitly allow arguments to ``()``, i.e. let me call ``(): hex"123456"`` and ``(), 1 ether: hex"123456"``. That will require a simple change to ``ExecutionFramework::callFallbackWithValue`` which will basically change it to ``ExecutionFramework::callLowLevel(u256 const& _value = u256(0), bytes const& _daa = {})``.

Reply:
*That's* actually what's needed to distinguish calls to ``ether`` and ``fallback`` - that's independent of whether you specify value, but rather depends on whether you send data or not.